OrbitGl/App.cpp

```diff
@@ -765,9 +765,12 @@ void OrbitApp::AddUiMessageCallback(
 
 //-----------------------------------------------------------------------------
 void OrbitApp::StartCapture() {
-  // Tracing session is only needed when StartCapture is
-  // running on the service side
-  Capture::StartCapture(options_.asio_server_address);
+  bool started = Capture::StartCapture(options_.asio_server_address);
+  if (!started) {
+    SendErrorToUi("No process selected",
```

Comment:
Could we move the error message to Capture::StartCapture for the case where false is returned if the process name is empty?  It seems it can return false in other scenarios as well.

Reply:
`Capture::StartCapture` doesn't know about `OrbitApp` so the way to do it would instead be to change the return type of `StartCapture`. I can do it, but I wasn't bothering as the other cause for `return false` is inside an `if (!IsRemote())` that should probably just be removed (the service should always be separate).

Reply:
I guess there are more situations like this. We could add a `Result<T>` type, having two subclasses, `Success` and `Failure`. Where the `Failure` carries an error message, and `Success` carries a `T` type value (could be void).
A bit like `Try` in Scala.

Reply:
I like the idea @florian-kuebler!  I'm also OK with implementing that in another PR.

Reply:
There are various implementations and proposals, under various names such as `expected`. None is standard and I'm not aware of one in Abseil, so we should think twice before jumping on one. It seems that in `OrbitSsh` we are using https://github.com/ned14/outcome, but the error is a `std::error_code`. I can also just return a `std::string` (empty for success) or a `std::optional<std::string>`.
@dimitry- @beckerhe Do you have suggestions or preferences?

Reply:
You can use outcome::result<void,string> - where string is error type.

Reply:
Ah, right, so it is possible to change the error type. So I tried this and&mdash;speaking of non-standard...&mdash;it causes compilation errors in llvm and protobuf with clang because it does something not very recommendable like
```
#if !defined(__cpp_sized_deallocation)
#define __cpp_sized_deallocation 190000
#endif
```

Reply:
(well.. maybe we should implement our own expected<>/StatusOr<>)

I think it is more readable if you added a std::string* error parameter to function call - this way if false is returned the error is set. Using optional here is counterintuitive, returning string is also not very expressive.

Reply:
So thanks to @beckerhe I got `outcome` to work. Do you still prefer `bool` + `std::string*` or should I go with `outcome`?

Reply:
if outcome works let's go with outcome for now.
what was the problem btw?

Reply:
clang does not enable the sized deallocation feature which was added in C++14 by default because it would break ABI.

outcome includes a library which was developed in WG21 SG-10 to polyfill the feature test macros which will be included in C++20 as part of the standard library (https://en.cppreference.com/w/cpp/feature_test). This library enables the feature test macro for sized-deallocation whenever the compilation mode is at C++14 or above. LLVM and protobuf check this feature test macro and enable sized-deallocation even though clang does not have it enabled.

outcome itself does not need sized-deallocation but since this polyfill-library is included in its entirety it leads to problems. Our proposed solution is to enable sized-deallocation in clang by the `-fsized-deallocation`-flag.